### PR TITLE
Testing Upgrade and Intensification

### PR DESCRIPTION
**Pull request by Trompettesib**

### CODE REVIEW

**Comment by Trompettesib** [comment]
I struggle with the cmake rn. If you have an idea on how to solve it, please suggest it on this pr

**Comment by Trompettesib** [comment]
I finally found the problem !

**Comment by Trompettesib** [comment]
If you need more tests, do not hesitate to ask me

**Review by STMiki — DISMISSED** [review]
Is there a reason you commented some tests ?

**Comment by Trompettesib** [comment]
> Is there a reason you commented some tests ?

Yep, because some of the functions does not have parsing and serializing

File: cubic-server/CMakeLists.txt [hunk 1]
```diff
@@ -42,14 +42,23 @@ if (GTEST)
...
         GTest::gtest_main
+        rapidcheck
+        rapidcheck_gtest
     )
 
+    # file(DOWNLOAD https://cdn.cubic-mc.com/tests/hello_world.nbt, ${CMAKE_CURRENT_BINARY_DIR}/hello_world.nbt SHOW_PROGRESS TLS_VERIFY ON)
+    # file(DOWNLOAD https://cdn.cubic-mc.com/tests/bigtest.nbt, ${CMAKE_CURRENT_BINARY_DIR}/bigtest.nbt SHOW_PROGRESS TLS_VERIFY ON)
+
+    execute_process(COMMAND curl -o ${CMAKE_CURRENT_BINARY_DIR}/hello_world.nbt https://cdn.cubic-mc.com/tests/hello_world.nbt)
+    execute_process(COMMAND curl -o ${CMAKE_CURRENT_BINARY_DIR}/bigtest.nbt https://cdn.cubic-mc.com/tests/bigtest.nbt)
```
Comment (huntears) [inline cubic-server/CMakeLists.txt:60]: That sounds scuffed

**Review by huntears — APPROVED** [review]

File: cubic-server/protocol/serialization/popPrimaryType.hpp [hunk 2]
```diff
@@ -110,15 +110,15 @@ constexpr int64_t popVarLong(uint8_t *&at, uint8_t *eof)
...
+        value |= ((uint64_t) (currentByte & SEGMENT_BITS)) << position;
 
         if ((currentByte & CONTINUE_BIT) == 0)
             return value;
 
         position += 7;
 
         if (position >= 64)
-            throw VarIntOverflow("VarInt exceeds max length");
+            throw VarIntOverflow("VarLong exceeds max length"); // TOOO(huntears): Change this to its proper exception
```
Comment (STMiki) [inline cubic-server/protocol/serialization/popPrimaryType.hpp:121]: That's literally a define 😆

**Review by STMiki — APPROVED** [review]

**Pull request merged**